File: Backend/obg.Domain/Entities/Pharmacy.cs
using obg.Domain.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Data;
using System.Text;

namespace obg.Domain.Entities
{
    public class Pharmacy
    {
        [Key] public string Name { get; set; }
        public string Address { get; set; }
        public List<Medicine> Medicines { get; set; }
        public List<Demand> Demands { get; set; }
        public List<Purchase> Purchases { get; set; }

        public Pharmacy() { }
        public Pharmacy(string name, string address)
        {
            Name = name;
            Address = address;
            Medicines = new List<Medicine>();
            Demands = new List<Demand>();
        }

        public override bool Equals(object obj)
        {
            if (obj == null)
            {
                return false;
            }
            Pharmacy pharmacy = obj as Pharmacy;
            if (pharmacy == null)
            {
                return false;
            }
            return this.Name == pharmacy.Name;
        }

    }
}

File: ObgApp/src/app/models/pharmacy.ts
import { Demand } from "./demand";
import { Medicine } from "./medicine";
import { Purchase } from "./purchase";

export class Pharmacy {
    name: string | null;
    address: string | null;
    medicines: Medicine[] | null;
    demands: Demand[] | null;
    purchases: Purchase[] | null;

    constructor(name: string | null, address: string | null, medicines: Medicine[] | null, demands: Demand[] | null, purchases: Purchase[] | null) {
        this.name = name;
        this.address = address;
        this.medicines = medicines;
        this.demands = demands;
        this.purchases = purchases;
    }
}